syntax = "proto3";

package brooks.dice.proto;

message DiceRollRequest {
  // Example: 2d4 + 1d20 + 5 => [ 2d4, 1d20, 5 ]
  repeated string roll = 1;
}

message DiceRollResponse {
  optional uint64 total = 1;
  repeated uint64 roll = 2;
}

message Dice {
  uint32 count = 1;
  uint32 sides = 2;
}

service DiceRollService {
  rpc Roll(DiceRollRequest) returns (DiceRollResponse);
}

